From 86904ab955c1d0d40b851bb28b25c3167b9eab61 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Tue, 5 Jan 2021 07:04:57 +0700
Subject: [PATCH 04/15] Fix linking with modern gtkmacintegration

---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 277935e..57e529b 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -641,7 +641,7 @@ sylpheed_LDADD = \
 	$(LDAP_LIBS) \
 	$(LIBICONV) \
 	$(SYLPHEED_LIBS) \
-	-lgtkmacintegration \
+	-lgtkmacintegration-gtk2 \
 	libsylpheed-plugin-0.la \
 	../libsylph/libsylph-0.la \
 	$(SYLPHEED_RES)
-- 
2.32.1 (Apple Git-133)

