From 748ab27d621244e401ec9cdd86b82d595f2c27d9 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 12 Jun 2022 03:41:00 +0700
Subject: [PATCH 13/17] Add more options available in macOS for Web browser

---
 src/prefs_common_dialog.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/prefs_common_dialog.c b/src/prefs_common_dialog.c
index f3c38df..321afa6 100644
--- a/src/prefs_common_dialog.c
+++ b/src/prefs_common_dialog.c
@@ -3021,6 +3021,19 @@ static GtkWidget *prefs_extcmd_create(void)
 	gtkut_combo_set_items (GTK_COMBO (uri_combo),
 #if defined(G_OS_WIN32) || defined(__APPLE__)
 			       _("(Default browser)"),
+#  if defined(__APPLE__)
+			       "open -a 'Safari' '%s'",
+			       "open -a 'Google Chrome' '%s'",
+			       "open -a 'Chromium' '%s'",
+			       "open -a 'Firefox' '%s'",
+			       "open -a 'Tor Browser' '%s'",
+			       "open -a 'Microsoft Edge' '%s'",
+			       "open -a 'Internet Explorer' '%s'",
+			       "open -a 'Opera' '%s'",
+			       "open -a 'Vivaldi' '%s'",
+			       "open -a 'Brave Browser' '%s'",
+			       "open -a 'Yandex' '%s'",
+#  endif
 #else
 			       DEFAULT_BROWSER_CMD,
 			       "sensible-browser '%s'",
-- 
2.32.1 (Apple Git-133)

