From dbfbe31ad2d29c8bebd5016ffe0c30207ee61c3b Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 13 Jun 2022 07:27:28 +0700
Subject: [PATCH 14/14] Add more options available in macOS for Editor

---
 src/prefs_common_dialog.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/prefs_common_dialog.c b/src/prefs_common_dialog.c
index 321afa6..7c8788f 100644
--- a/src/prefs_common_dialog.c
+++ b/src/prefs_common_dialog.c
@@ -3062,6 +3062,20 @@ static GtkWidget *prefs_extcmd_create(void)
 			       "notepad '%s'",
 #elif defined(__APPLE__)
 			       "open -n -W -e '%s'", /* TextEdit */
+			       "'/Applications/MacVim.app/Contents/bin/mvim' -g --nofork -c 'au VimLeave * maca hide:' '%s'",
+			       "open -n -W -a 'Emacs' --args '%s'",
+			       "open -n -W -a 'gedit' --args --new-window --wait --standalone '%s'",
+			       "open -n -W -a 'kate' --args --startanon --new '%s'",
+			       "open -n -W -a 'CudaText' --args -n -nh -ns '%s'",
+			       "'/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code' --new-window --wait '%s'",
+			       "'/Applications/VSCodium.app/Contents/Resources/app/bin/codium' --new-window --wait '%s'",
+			       "'/Applications/Atom.app/Contents/Resources/app/atom.sh' --new-window --wait '%s'",
+			       "'/Applications/TextMate.app/Contents/MacOS/mate' --wait '%s'",
+			       "xed --wait '%s'", /* Xcode */
+			       "open -n -W -a 'Brackets' '%s'",
+			       "'/Applications/Sublime Text.app/Contents/SharedSupport/bin/subl' --new-window --wait '%s'",
+			       "'/Applications/BBEdit.app/Contents/Helpers/bbedit_tool' --wait --new-window --resume '%s'",
+			       "'/Applications/TextWrangler.app/Contents/Helpers/edit' --wait --new-window --resume '%s'",
 #else
 			       "gedit %s",
 			       "kedit %s",
-- 
2.32.1 (Apple Git-133)

