From 6506b885fa34679df2c35bc281ea35fb8ee8129b Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 26 Sep 2022 09:33:09 +0700
Subject: [PATCH] GLIB: disable utimensat

---
 gio/glocalfileinfo.c | 2 ++
 meson.build          | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/gio/glocalfileinfo.c b/gio/glocalfileinfo.c
index 6f18695..c9e5c8b 100644
--- a/gio/glocalfileinfo.c
+++ b/gio/glocalfileinfo.c
@@ -2800,6 +2800,7 @@ set_mtime_atime (char                       *filename,
       return FALSE;
     }
 
+#if defined (HAVE_UTIMENSAT)
   res = utimensat (AT_FDCWD, filename, times_n, 0);
   if (res == -1)
     {
@@ -2811,6 +2812,7 @@ set_mtime_atime (char                       *filename,
                    g_strerror (errsv));
       return FALSE;
     }
+#endif
   return TRUE;
 }
 #endif
diff --git a/meson.build b/meson.build
index 3aa1a10..d7118fd 100644
--- a/meson.build
+++ b/meson.build
@@ -635,7 +635,8 @@ functions = [
   'unsetenv',
   'uselocale',
   'utimes',
-  'utimensat',
+  # warning: 'utimensat' is only available on macOS 10.13 or newer [-Wunguarded-availability-new]
+  # 'utimensat',
   'valloc',
   'vasprintf',
   'vsnprintf',
-- 
2.37.0 (Apple Git-136)

