From 96080d790c8f6efee380e9ae1f042a9e142c7748 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 26 Sep 2022 08:59:50 +0700
Subject: [PATCH] XZ: disable futimens

---
 config.h.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/config.h.in b/config.h.in
index 33a99df..9efe50b 100644
--- a/config.h.in
+++ b/config.h.in
@@ -154,7 +154,10 @@
 #undef HAVE_FUNC_ATTRIBUTE_CONSTRUCTOR
 
 /* Define to 1 if you have the 'futimens' function. */
+#if 0
+/* warning: 'futimens' is only available on macOS 10.13 or newer [-Wunguarded-availability-new] */
 #undef HAVE_FUTIMENS
+#endif
 
 /* Define to 1 if you have the 'futimes' function. */
 #undef HAVE_FUTIMES
-- 
2.49.0

