From a40a01f719eb1ca24a72b08fe2744cba0cc0fb38 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Tue, 19 Aug 2025 01:19:23 +0700
Subject: [PATCH 15/33] Fix multilanguage spell checking

---
 src/compose.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/compose.c b/src/compose.c
index 42838fd..6980cd3 100644
--- a/src/compose.c
+++ b/src/compose.c
@@ -6283,7 +6283,9 @@ static void compose_set_spell_lang_menu(Compose *compose)
 	enchant_broker_list_dicts(eb, ench_dict_desc_cb, &dict_list);
 	enchant_broker_free(eb);
 
-	for (cur = dict_list; cur != NULL; cur = cur->next) {
+	if (compose->spell_lang != NULL && strchr(compose->spell_lang, ','))
+		lang_set = TRUE;
+	for (cur = dict_list; cur != NULL && !lang_set; cur = cur->next) {
 		if (compose->spell_lang != NULL &&
 		    g_ascii_strcasecmp(compose->spell_lang,
 				       (gchar *)cur->data) == 0)
-- 
2.53.0

