From ddf7d5652558ee77bf804350f13dc8ef6b098816 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Tue, 14 Jun 2022 18:15:22 +0700
Subject: [PATCH 28/34] Allow forwarding configure args

---
 makeosx.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/makeosx.sh b/makeosx.sh
index 92ae1ed..0671d94 100755
--- a/makeosx.sh
+++ b/makeosx.sh
@@ -1,6 +1,7 @@
 #!/bin/sh
 
 ./configure --prefix=$HOME/gtk/inst \
+  "${@}" \
   --with-localedir=Contents/Resources/share/locale \
   --with-themedir=Contents/Resources/share/icons \
   && make \
-- 
2.53.0

